ports/sfml/portfile.cmake

```
@@ -32,9 +32,11 @@ vcpkg_install_cmake()
 vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/SFML)
 vcpkg_copy_pdbs()
 
+FILE(READ ${CURRENT_PACKAGES_DIR}/share/sfml/SFMLConfig.cmake SFML_CONFIG)
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
-    FILE(READ ${CURRENT_PACKAGES_DIR}/share/sfml/SFMLConfig.cmake SFML_CONFIG)
     FILE(WRITE ${CURRENT_PACKAGES_DIR}/share/sfml/SFMLConfig.cmake "set(SFML_STATIC_LIBRARIES true)\ninclude(CMakeFindDependencyMacro)\nfind_dependency(Freetype)\n${SFML_CONFIG}")
+else()
+    FILE(WRITE ${CURRENT_PACKAGES_DIR}/share/sfml/SFMLConfig.cmake "set(SFML_STATIC_LIBRARIES false)\n${SFML_CONFIG}")
```

I don't get why you need this change. I also don't get why `imgui-sfml ` required `SFML_STATIC_LIBRARIES`
since SFML seems to already set `SFML_STATIC_LIBRARIES` if required.

---

There is potentially an issue in the reverse case -- if `sfml` is dynamic and `imgui-sfml` is static. This change ensures that `SFML_STATIC_LIBRARIES` is always set exactly according to how sfml was built, regardless of whether the outer project sets the variable or not.

---

Better make that a `FATAL_ERROR` instead of silently accepting it because it means a consumer needs to be patched instead and might be assuming something wrongly.

---

I think that it's better to be optimistic here:
1. Patching every single consumer is not realistic
2. The vast majority of cases will work without patching
3. The cases that don't work will have build failures, not silent misbehavior